### PR TITLE
Landing Page

### PR DESCRIPTION
**Pull request by jheel05**
This PR contains the Static Changes of the landing page, i.e, NavBar, CategoriesBar, Image carousel, Footer.

### CODE REVIEW

**Comment by jheel05** [comment]
The PR comments have been addressed, and changed the file structure.

**Comment by srivsaks** [comment]
Please add screenshots/video recoridng of the changes

File: src/App.jsx [hunk 1]
```diff
@@ -1,13 +1,11 @@
-
-import './App.css'
+import LandingPage from './pages/landingpage.jsx'
```
Comment (srivsaks) [inline src/App.jsx]: Install prettier and run that on all your files.

File: src/components/Login/login.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
```
Comment (srivsaks) [inline src/components/Login/login.jsx]: Remove this file from the PR if its not ready or else finish this one and push the new changes in the next round of revision.

File: src/components/ImageSlider/imageslider.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import { useEffect } from "react";
+import { images } from "./imageSlider";
+
+
+function Slider() {
+
+  const [currentImage, setCurrentImage] = useState(0); 
+  const [isHovering, setIsHovering] = useState(false);
+  const nextImage = () => {
+    setCurrentImage((prevImage) => (prevImage + 1) % images.length);
```
Comment (srivsaks) [inline src/components/ImageSlider/imageslider.jsx]: images.length could have been a variable and then used at places.

File: src/components/ImageSlider/imageslider.jsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      intervalTimer = setInterval(() => {
+        nextImage();
+      }, 2500);
+    }
+    
+
+    return()=>{
+      clearInterval(intervalTimer);
+    }
+  })
```
Comment (srivsaks) [inline src/components/ImageSlider/imageslider.jsx]: why no dependency? do we want to make this useEffect run on every render?

File: src/components/ImageSlider/imageslider.jsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+          className={`inline-block h-1 w-2 rounded-full ${currentImage === index ? 'bg-white' : 'bg-gray-500'}`}
+        ></span>
+      ))}
+    </div>
+
+      {/* Slider button container */}
+      <div className='w-full h-fit flex justify-between absolute left-0 top-[50%] translate-y-[-50%]'>
+        <span onClick={prevImage} className='bg-white shadow-[0_3px_10px_rgb(0,0,0,0.2)] px-[2px] py-3 sm:px-1 sm:py-5 lg:px-3 lg:py-9 rounded-r-sm cursor-pointer'><FaAngleLeft /></span>
+        <span onClick={nextImage} className='bg-white shadow-[0_3px_10px_rgb(0,0,0,0.2)] px-[2px] py-3 sm:px-1 sm:py-5 lg:px-3 lg:py-9 rounded-l-sm cursor-pointer'><FaAngleRight /></span>
+      </div>
```
Comment (srivsaks) [inline src/components/ImageSlider/imageslider.jsx]: Convert it to a reusable component and then use it.

Comment (srivsaks) [reply src/components/ImageSlider/imageslider.jsx]: Also why not button element? why a span element ?

File: src/components/Footer/footer.jsx [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+      <div id="footer">
+        <div>
+          <li className="heading">ABOUT</li>
+          <li>
+            <a
+              href="https://corporate.flipkart.net/corporate-home"
+              target="_blank"
+              >About Us</a
+            >
+          </li>
```
Comment (srivsaks) [inline src/components/Footer/footer.jsx]: Make this a re-usable component.Applicable at other places too.

File: src/utils/search.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+let timer;
```
Comment (srivsaks) [inline src/utils/search.js]: convert it into react code and not js.

Comment (jheel05) [reply src/utils/search.js]: Resolved.

File: src/components/ImageSlider/imageslider.jsx (hunk 4 above)
Comment (jheel05) [reply src/components/ImageSlider/imageslider.jsx]: Resolved

File: src/components/ImageSlider/imageslider.jsx (hunk 3 above)
Comment (jheel05) [reply src/components/ImageSlider/imageslider.jsx]: Resolved

File: src/components/ImageSlider/imageslider.jsx (hunk 5 above)
Comment (jheel05) [reply src/components/ImageSlider/imageslider.jsx]: Resolved

File: src/components/Footer/footer.jsx (hunk 6 above)
Comment (jheel05) [reply src/components/Footer/footer.jsx]: Resolved

File: src/App.jsx (hunk 1 above)
Comment (jheel05) [reply src/App.jsx]: Resolved

**Comment by jheel05** [comment]
Updated the Landing Page. Addressed all previous comments on the PR. Now includes header, footer, product list display by categories(using API), image carousel, search function(using API), and login functionality(using API). Check it out!

Screenshots:
Screenshot 2024-03-06 at 2 07 54 PM
Screenshot 2024-03-06 at 2 07 57 PM
Screenshot 2024-03-06 at 2 08 00 PM
Screenshot 2024-03-06 at 2 08 08 PM
Screenshot 2024-03-06 at 2 08 15 PM
Screenshot 2024-03-06 at 2 08 24 PM
Screenshot 2024-03-06 at 2 08 35 PM

File: src/components/Login/login.jsx (hunk 2 above)
Comment (jheel05) [reply src/components/Login/login.jsx]: Resolved

File: src/API/CategoryProducts/CategoryProductAPI.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+// api.js
```
Comment (srivsaks) [inline src/API/CategoryProducts/CategoryProductAPI.js:1]: nit: you can remove this.

Comment (srivsaks) [inline src/API/CategoryProducts/CategoryProductAPI.js:1]: give folder structure name src/api ....

Comment (jheel05) [reply src/API/CategoryProducts/CategoryProductAPI.js:1]: Resolved

Comment (jheel05) [reply src/API/CategoryProducts/CategoryProductAPI.js:1]: Resolved

**Pull request merged**